src/uwtools/tests/drivers/test_driver.py

```diff
@@ -82,6 +100,153 @@ def schema():
 """.strip()
 
 
+def test_create_directory_structure(tmp_path):
+    run_dir = tmp_path / "run"
+    with patch.object(Driver, "_create_run_directory") as _create_run_directory:
+        Driver.create_directory_structure(run_dir)
+    _create_run_directory.assert_called_once_with(run_dir, "delete")
+
+
+def test_create_directory_structure_bad_existing_act():
+    with raises(ValueError):
+        Driver.create_directory_structure(run_directory="/some/path", exist_act="foo")
+
+
+def test_run_cmd_expected(mpi_config, tmp_path):
+    config_file = tmp_path / "config.yaml"
+    YAMLConfig.dump_dict(config_file, mpi_config)
+    with patch.object(Driver, "_validate", return_value=True):
+        driver = ConcreteDriver(config_file=config_file)
+    driver._config = driver._experiment_config.get("component", {})
+    assert driver.run_cmd() == "srun bar foo"
+
+
+def test_run_cmd_no_runtime_info(mpi_config, tmp_path):
+    config_file = tmp_path / "config.yaml"
+    del mpi_config["component"]["runtime_info"]
+    YAMLConfig.dump_dict(config_file, mpi_config)
+    with patch.object(Driver, "_validate", return_value=True):
+        driver = ConcreteDriver(config_file=config_file)
+    driver._config = driver._experiment_config.get("component", {})
+    with raises(KeyError):
+        driver.run_cmd()
+
+
+def test_scheduler():
+    config_file = fixture_path("fruit_config.yaml")
+    with patch.object(Driver, "_validate", return_value=True):
+        concretedriver = ConcreteDriver(config_file=config_file)
+        # pylint: disable=pointless-statement
+        with patch.object(driver.JobScheduler, "get_scheduler") as _get_scheduler:
```

Comment:
Why the apparent private name for the `_get_scheduler` mock (that is, vs just `get_scheduler`)? If we _did_ need a different name for a local variable, e.g. to avoid conflicting with a Python keyword, I think the idiom would be an underscore suffix rather than a prefix.

Reply:
That's the reason for a lot of weird things I do. That and ignorance.